frontend/components/FileBrowser/index.tsx

```diff
@@ -55,58 +47,55 @@ export default function Browser({ bucket }: Props) {
         hasAccess(apiContext, `fileHandler:${bucket}:delete`) && ChonkyActions.DeleteFiles,
     ];
 
-        useFilesQuery({
-            variables: {
-                bucket: hasAccess(apiContext, `fileHandler:${bucket}:read`) && bucket,
-                prefix: currentPath,
-            },
-            fetchPolicy: 'no-cache',
-            onCompleted: (data) => {
-                setFiles(data.files);
-            }
-        });
-    
-
+    useFilesQuery({
+        variables: {
+            bucket: hasAccess(apiContext, `fileHandler:${bucket}:read`) && bucket,
```

Comment:
Isn't it better to only run the query if you have access? `if (hasAccess(...)) { useFilesQuery(...`